src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -269,6 +278,30 @@ private static Uri CreateConnectUrl(Uri url, NegotiationResponse negotiationResp
             }
         }
 
+        private TransferMode GetTransferMode()
+        {
+            var transferModeFeature = Features.Get<ITransferModeFeature>();
+            if (transferModeFeature == null)
+            {
+                // REVIEW: use TransferMode.Text as default if the feature is missing?
+                throw new InvalidOperationException("TransferModeFeature not set.");
+            }
+
+            return transferModeFeature.TransferMode;
+        }
+
+        private void SetTransferMode(TransferMode transferMode)
```

Comment:
The feature, why does this method throw if the feature isn't there instead of just adding the feature on the fly.